docs/api/cluster/icdm.rst

```diff
@@ -0,0 +1,35 @@
+.. -*- mode: rst -*-
+
+Intercluster Distance Maps
+==========================
+
+Intercluster distance maps display an embedding of the cluster centers in 2 dimensions with the distance to other centers preserved. E.g. the closer to centers are in the visualization, the closer they are in the original feature space. The clusters are sized according to a scoring metric. By default, they are sized by membership, e.g. the number of instances that belong to each center. This gives a sense of the relative importance of clusters. Note however, that because two clusters overlap in the 2D space, it does not imply that they overlap in the original feature space.
+
```

Comment:
Yeh, that is a good resource - I read it while implementing this and we should reference it in the docs. I would very much like to extend the documentation with real-life examples, I think the `make_blobs` example is a good first example just to show how it works, but I'm hoping we can add an LDA example with text and the walking path example soon. 

I'd also like to include in the documentation a detailed description of how the legend works since it might not be obvious right off the bat. The three circles represent the 25th, 50th, and 75th percentile sizes of the circles drawn on the graph (scaled to the inset size).  I think having documentation about how to interpret the legend (e.g. what it means that the median is close to the 75th percentile, etc.) would also enhance the documentation.